api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java

```diff
@@ -209,6 +210,30 @@ void shouldCreateInRestriction() {
     @DisplayName("should create NotIn constraint correctly")
     @Test
     void shouldCreateNotInRestriction() {
+        var restriction = (BasicRestriction<Book, String>) _Book.author.in("Alice", "Bob").negate();
+
+        SoftAssertions.assertSoftly(soft -> {
+            soft.assertThat(restriction.expression()).isEqualTo(_Book.author);
+            soft.assertThat(restriction.constraint()).isEqualTo(NotIn.values("Alice", "Bob"));
+            soft.assertThat(restriction.constraint()).isInstanceOf(NotIn.class);
+        });
+    }
+
+    @DisplayName("should create In, using Set, constraint correctly")
+    @Test
+    void shouldCreateInAsSetRestriction() {
+        var restriction = (BasicRestriction<Book, String>) _Book.author.in(List.of("Alice", "Bob"));
+
+        SoftAssertions.assertSoftly(soft -> {
+            soft.assertThat(restriction.expression()).isEqualTo(_Book.author);
+            soft.assertThat(restriction.constraint()).isEqualTo(In.values("Alice", "Bob"));
+            soft.assertThat(restriction.constraint()).isInstanceOf(In.class);
+        });
+    }
+
+    @DisplayName("should create NotIn, using Set, constraint correctly")
+    @Test
+    void shouldCreateNotInAsSetRestriction() {
         var restriction = (BasicRestriction<Book, String>) _Book.author.in(List.of("Alice", "Bob")).negate();
```

Comment:
A test that explicitly says it is using `Set` should actually use `Set`,
```suggestion
    @DisplayName("should create NotIn, using Set, constraint correctly")
    @Test
    void shouldCreateNotInAsSetRestriction() {
        var restriction = (BasicRestriction<Book, String>) _Book.author.in(Set.of("Alice", "Bob")).negate();
```